### PR TITLE
Use github action to publish snap

### PR DESCRIPTION
**Pull request by LyzardKing**
Use github actions to publish,
to try and reduce the upload time

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not: Issue created at <https://github.com/JabRef/user-documentation/issues>.

### CODE REVIEW

**Comment by LyzardKing** [comment]
@Siedlerchr The upload is taking a lot longer than it should..
Maybe the "semi-official" github action can help..

**Comment by Siedlerchr** [comment]
Please note that the snap craft upload is currently only for master branch, at least I saw a filter

**Comment by koppor** [comment]
I don't know which action @Siedlerchr is referring to. Maybe you can add a link?

@LyzardKing I would ❤️ if you added an [ADR](https://adr.github.io/) for the choice of the action. Just place it at https://github.com/JabRef/jabref/tree/master/docs/adr.

I found following action as alternative - can just be added to the list of options ;) --> https://github.com/marketplace/actions/snapcraft-action

**Comment by LyzardKing** [comment]
The GitHub actions are from https://forum.snapcraft.io/t/call-for-testing-github-action-for-snapcraft/14930
They are developed from someone in the snapcraft team.
At the moment we're already using his build action.
This pr adds the upload action

File: .github/workflows/deployment.yml [hunk 1]
```diff
@@ -117,11 +117,11 @@ jobs:
       - name: Build snap (1) Setup snapcraft
```
Comment (Siedlerchr) [inline .github/workflows/deployment.yml]: I think it would make sense to run this in parallel as it's own job?

Comment (koppor) [reply .github/workflows/deployment.yml]: Currently does not work at all: https://github.com/JabRef/jabref/actions/runs/49715346

    cannot read snap file "build/distribution/jabref*.snap"

Trying to debug.

Comment (koppor) [reply .github/workflows/deployment.yml]: It's also very difficult to do it in parallel in a separate workflow (which is not what you meant, right?). IMHO it depends on the steps above (including jpackage, ...). Thus, we would need to duplicate 100 SLOC of the build file. 

We could work with different jobs in the same build file. Then, we would need to introduce uploading the `build` directory (more?) to GitHub as data passing between jobs can only be done that way. This is not comfortable (we have to be sure that we don't forget something)

Comment (koppor) [reply .github/workflows/deployment.yml]: Cannot read is because the `*` is not expanded. Fixed it in cd707467ab5ffb7d5c00

**Comment by LyzardKing** [comment]
Thanks both for looking at this..
I didn't have much time to do this while on a train with no stable internet connection.

**Comment by Siedlerchr** [comment]
Both master and the master-release fail at some point in the snapcraft upload
##[error]The operation was canceled. 
@LyzardKing  Btw, we still also have the snapcraft.yml , so I could try to trigger a rebuild there at the official snapcraft ci if that is possible with our infrastructure

**Comment by LyzardKing** [comment]
@koppor I changed the snap to use https://builds.jabref.org/master/JabRef-5.0-portable_linux.tar.gz instead of the build artifact (you can see the comment in the snapcraft.yaml) to test it..
If we keep the source as https://builds.jabref.org/master/JabRef-5.0-portable_linux.tar.gz it might be easier to add it to the snapcraft CI and use it there (it should rebuild every time a new tar.gz is uploaded.
If we want to use the github action instead is it possible to make the snap action run after the artifact build action?

**Comment by LyzardKing** [comment]
@Siedlerchr I think it might be easier if we try and run the snapcraft CI..
The snapcraft as it is now (with the full url as source) will work with that.
If it works then we can remove the snap part in the github actions, it seems to complicated to get working properly..

**Comment by Siedlerchr** [comment]
[Snap Status](https://build.snapcraft.io/user/JabRef/jabref)
I triggered a new build

**Comment by Siedlerchr** [comment]
@LyzardKing  Snapcraft runs but shows this weird message regarding license:

https://build.snapcraft.io/user/JabRef/jabref/858770
```
Could not find '/snap/core/current/usr/bin/snap', validation of the license string will only take place once pushed to the store.
Snapping 'jabref' ...

Snapped jabref_5.0.30532_amd64.snap
Revoking proxy token...
RUN: /usr/share/launchpad-buildd/bin/in-target scan-for-processes --backend=lxd --series=bionic --arch=amd64 SNAPBUILD-858770
Scanning for processes to kill in build SNAPBUILD-858770
```

**Comment by LyzardKing** [comment]
@Siedlerchr The review is manual and will need roughly two weeks.
https://forum.snapcraft.io/t/system-files-under-mozilla-native-messaging-hosts/13627/41

**Comment by Siedlerchr** [comment]
Ah okay thanks for checking that!

**Comment by koppor** [comment]
@LyzardKing The "manual review" is for each published version or a general "OK" for this and all upcoming next versions?

**Comment by LyzardKing** [comment]
As I understand it the approval should be done once for be permissions we asked.
So unless we change the permissions it shouldn't be done any more

**Pull request merged**